src/main/scala/decider/Decider.scala

```diff
@@ -34,6 +35,7 @@ trait Decider {
   def checkSmoke(): Boolean
 
   def setCurrentBranchCondition(t: Term): Unit
+  def setCurrentBranchConditionWithExp(t: Term, te: Exp): Unit
```

Comment:
On second thought: if callers typically need two separate stacks, might be better to internally record `Term` separately from `Option[Exp]`, i.e. have two `private var`s instead of one, but to only allow setting both at ones.

Reply:
Good point, thanks! I think I'll keep two stacks but only allow pushing to both at once.